### PR TITLE
HPCC-21997 Combine duplicate LDAP Security methods getResources

### PR DESCRIPTION
**Pull request by RussWhitehead**
Currently, in LDAP Security Manager, the 2 methods getResouces and getResourcesEx
are almost identical. This PR ccombines them into a single getResources call,
which handles the functionality of both. Note that an apparent bug is also
fixed that was previously in getResourcesEx, where the GetAttributesWrapper
looped through all attributes before saving, which meant only the last one was
saved. Also, pass the LDAP connection into the getManagedScopeTree method since
it is available

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@wangkx Please review. Note the main changes were to combine getResources/getResourcesEx into a single function

File: system/security/LdapSecurity/ldapconnection.cpp [hunk 1]
```diff
@@ -3461,88 +3461,26 @@ class CLdapClient : implements ILdapClient, public CInterface
...
-
-    virtual bool getResourcesEx(SecResourceType rtype, const char * basedn, const char* prefix, const char* searchstr, IArrayOf<ISecResource>& resources)
-    {
         if(rtype == RT_FILE_SCOPE || rtype == RT_WORKUNIT_SCOPE)
         {
             assertex(isEmptyString(searchstr));
             assertex(isEmptyString(prefix));
 
-            getManagedScopeTree(rtype, basedn, resources);
+            getManagedScopeTree(ld, rtype, basedn, resources);//@@TODO Pass LD to getManagedScopeTree
```
Comment (wangkx) [inline system/security/LdapSecurity/ldapconnection.cpp]: TODO?

**Review by wangkx — COMMENTED** [review]
@RussWhitehead 2 comments

File: system/security/LdapSecurity/ldapconnection.cpp (hunk 1 above)
Comment (RussWhitehead) [reply system/security/LdapSecurity/ldapconnection.cpp]: Oops, already did it

**Comment by RussWhitehead** [comment]
@wangkx Please rereview

**Review by wangkx — APPROVED** [review]
@RussWhitehead looks fine now.

**Comment by RussWhitehead** [comment]
@richardkchapman Squashed, please merge

**Comment by richardkchapman** [comment]
@RussWhitehead should be targetting 7.2.x or (probably) master

**Comment by RussWhitehead** [comment]
@richardkchapman I can't build master, so targeted 7.2.x. Please merge

**Comment by richardkchapman** [comment]
Why can't you build master? And that's NOT a good reason for selecting the target version of a fix!

File: version.cmake [hunk 2]
```diff
@@ -5,7 +5,7 @@ set ( HPCC_NAME "Community Edition" )
 set ( HPCC_PROJECT "community" )
 set ( HPCC_MAJOR 7 )
 set ( HPCC_MINOR 2 )
-set ( HPCC_POINT 3 )
```
Comment (richardkchapman) [inline version.cmake]: This change should not be included

**Comment by richardkchapman** [comment]
Needs rebasing

**Comment by richardkchapman** [comment]
@RussWhitehead

**Pull request closed without merging**